### DISCUSSION TITLE
Issues running BWA-MEM in a docker container

### DISCUSSION THREAD

**Discussion by Fyodir** [nextflow-io/nextflow#3691] [Q&A]
HI Everyone,

I am currently attempting to get BWA-MEM to run inside an isolated container within nextflow. Currently I can get nextflow to run, no problem at all with BWA installed locally on my Ubuntu 20.04 machine. However, Each time I attempt to run this process inside a container (which has an ubuntu 20.04 base image and BWA installed on that) Nextflow keeps informing me that it is unable to find thew BWA index files.

First I thought this may be a permissions issue with regards to the indexes. But that does not seem to be the case as BWA-MEM runs perfectly with the indexes outside of the container with a locally installed BWA. I am currently thinking that as it is running in a docker container, the location of the indexes are not within the scope of the BWA container?

Andy help would be greatly appreciated

nextflow.config
```groovy
params {
    dirData = false
    dirOut = false
    bwaThreads = 16
    refFasta = "/mnt/F_Drive/03_Reference/hg38/hg38.fa"
    haplotypeCallerMappingQual = 20
    haplotypeCallerMinBaseQual = 20
}

process {
    withName:fastQC {
        container = "mqmnd/fastqc:amd-0.11.9"
    }
    withName:multiQC {
        container = "mqmnd/multiqc:amd-1.12"
    }
    withName:bwamem {
        container = "mqmnd/bwa:amd-0.7.17-r1188"
    }
}

docker {
    enabled = true
    remove = true
}
```

```groovy
process bwamem {

	// publishDir path: "${params.dirOut}/alignment", pattern: "*sam", mode: "copy"

    input:
        tuple(val(sampleName), path(fastq))
        
    output:
        tuple(val(sampleName), path("*.bam"))

    script:
        """
        bwa mem \
        -t ${params.bwaThreads} \
        ${params.refFasta} \
        ${fastq[0]} ${fastq[1]} \
        -M | samtools sort -@ ${params.bwaThreads} - -o ${sampleName}.sorted.bam
        """
}
```

```groovy
// main workflow
workflow {

    // Pipeline input files
    Channel.fromFilePairs("${params.dirData}/RawData/*_R{1,2}.fastq.gz").set{ch_fastq_pairs}
    ch_fastq_pairs.view()

    // // Align fastqs to Human Genome, output a sorted BAM
    bwamem(ch_fastq_pairs)

}
    
```

**Comment by bentsherman** [comment]
If the index files aren't already available in the container image, you have to declare a process input for the index files and provide them when you call the process in your workflow.

**Comment by Fyodir** [comment]
Hi Ben,

Thank you very much for your reply. I will give that a go as it has been
driving me crazy!

Bw

On Thu, 2 Mar 2023 at 02:03, Ben Sherman ***@***.***> wrote:

> If the index files aren't already available in the container image, you
> have to declare a process input for the index files and provide them when
> you call the process in your workflow.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nextflow-io/nextflow/discussions/3691#discussioncomment-5168356>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKNAIIEDNT6A3WMU6H7TE2LWZ5XOXANCNFSM6AAAAAAVI64NGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Best Wishes,

Andrew Smith
